### PR TITLE
[DC-82] add metric for request access

### PR DESCRIPTION
**Pull request by tcjiang9**
separate metrics for opening the modal and actually requesting access

### CODE REVIEW

File: src/pages/library/RequestDatasetAccessModal.js [hunk 1]
```diff
@@ -60,7 +61,7 @@ export const RequestDatasetAccessModal = ({ onDismiss, datasets }) => {
           ]),
           td([
             Utils.switchCase(access,
-              ['Controlled', () => h(RequestDatasetAccessButton, { datasetName: title })],
+              ['Controlled', () => h(RequestDatasetAccessButton, datasets[0] )],
```
Comment (lu-c) [inline src/pages/library/RequestDatasetAccessModal.js]:
```suggestion
              ['Controlled', () => h(RequestDatasetAccessButton, { title, id } )],
```
This way it will change with which button the user clicks, so you can make requests for all of the datasets

File: src/pages/library/DataBrowserDetails.js [hunk 2]
```diff
@@ -198,7 +204,8 @@ const DataBrowserDetails = ({ id }) => {
           h(Sidebar, { snapshot, id, setShowRequestAccessModal }),
           showRequestAccessModal && h(RequestDatasetAccessModal, {
             datasets: [snapshot],
-            onDismiss: () => setShowRequestAccessModal(false)
+            onDismiss: () => setShowRequestAccessModal(false),
+            isSnapshot: true
```
Comment (lu-c) [inline src/pages/library/DataBrowserDetails.js]: Why did you add this field?

**Comment by kyuksel** [comment]
@tcjiang9 UI PRs need to have a ticket number in the PR title for compliance reasons.

File: src/pages/library/DataBrowserDetails.js (hunk 2 above)
Comment (tcjiang9) [reply src/pages/library/DataBrowserDetails.js]: accident, forgot to delete

File: src/pages/library/RequestDatasetAccessModal.js [hunk 3]
```diff
@@ -71,22 +72,26 @@ export const RequestDatasetAccessModal = ({ onDismiss, datasets }) => {
...
-
   return h(ButtonPrimary, {
     disabled: requesting || requested,
-    onClick: requestAccess
+    onClick: () => {
+      _.flow(
+        Utils.withBusyState(setRequesting),
+        withErrorReporting('Error requesting dataset access')
+      )(async () => {
+        await Ajax(signal).DataRepo.requestAccess(dataset.id)
```
Comment (petesantos) [inline src/pages/library/RequestDatasetAccessModal.js]: Does this need to be feature flagged?

File: src/pages/library/RequestDatasetAccessModal.js [hunk 4]
```diff
@@ -71,22 +72,26 @@ export const RequestDatasetAccessModal = ({ onDismiss, datasets }) => {
   ])
 }
 
-const RequestDatasetAccessButton = ({ dataset }) => {
+const RequestDatasetAccessButton = (dataset) => {
```
Comment (petesantos) [inline src/pages/library/RequestDatasetAccessModal.js]: Being a react component `dataset` is essentially the list of props being passed in to this component. We destructure the props on the component. Also, using Christine's suggestion above, this could be changed to `({title, id}) => {`
I would prefer this approach in the spirit of using the least powerful abstraction / simplicity. Given the amount of data in the dataset, I would prefer only passing in the necessary information for a small component like this.

File: src/pages/library/DataBrowserDetails.js (hunk 2 above)
Comment (lu-c) [reply src/pages/library/DataBrowserDetails.js]: oh haha, been there :) Can we delete it?

File: src/pages/library/RequestDatasetAccessModal.js (hunk 3 above)
Comment (tcjiang9) [reply src/pages/library/RequestDatasetAccessModal.js]: not sure what you mean, @lu-c ?

File: src/pages/library/DataBrowser.js [hunk 5]
```diff
@@ -173,6 +173,7 @@ const makeDataBrowserTableComponent = ({ sort, setSort, selectedData, toggleSele
       useHover: false,
       underRowKey: 'underRow',
       rows: _.map(datum => {
+        datum['access'] = 'Controlled'
```
Comment (lu-c) [inline src/pages/library/DataBrowser.js]: Please remove this line! Phil has a code change to set some of the datasets to Controlled

File: src/pages/library/DataBrowserDetails.js [hunk 6]
```diff
@@ -199,7 +205,7 @@ const DataBrowserDetails = ({ id }) => {
           h(Sidebar, { snapshot, id, setShowRequestAccessModal }),
           showRequestAccessModal && h(RequestDatasetAccessModal, {
             datasets: [snapshot],
-            onDismiss: () => setShowRequestAccessModal(false)
+            onDismiss: () => setShowRequestAccessModal(false),
```
Comment (lu-c) [inline src/pages/library/DataBrowserDetails.js]: The linter is complaining about this line having a comma dangle

**Review by lu-c — APPROVED** [review]
approved pending changes

File: src/pages/library/RequestDatasetAccessModal.js (hunk 3 above)
Comment (lu-c) [reply src/pages/library/RequestDatasetAccessModal.js]: I think pete is asking, "Does it need a "is dataCatalog enabled" check before you run this request?"

The requestAccess endpoint doesnt exist ATM, so this request will just fail. That error is being caught and showing "error requesting dataset access" to the user. I don't think we need to feature flag this, esp since access to this page should be feature flagged, but if there is something we should do, let me know!

**Review by petesantos — APPROVED** [review]
